testing-tools/testing-tools-macros/src/randomizer.rs

```diff
@@ -0,0 +1,171 @@
+                {
+                    //#random_field!($($field_name:ident, $value:expr),*)
+                    let mut rng = rand::thread_rng();
+                    let mut this: #field = rand::Rng::gen(&mut rng);
+                    $(
+                        this.$field_name = $value;
+                    )*
+                    this
+                }
+            };
+
+            (#variant_name) => {
+                {
+                    //#random_field!()
+                    let mut rng = rand::thread_rng();
+                    let this: _ = rand::Rng::gen(&mut rng);
+                    this
+                }
+            };
+        }
+    } else {
+        panic!("enum variant fields can only have 1 or 0 values");
+    }
+}
+
+fn any_enum_variant(enum_name: &Ident, variant_names: Vec<Ident>) -> proc_macro2::TokenStream {
+    let num_variants = (0..variant_names.len()).into_iter().collect::<Vec<_>>();
+
+    quote::quote! {
+        () => {
+            {
+                let mut rng = rand::thread_rng();
+                let this: usize = rand::Rng::gen(&mut rng);
+                match this {
+                    #(
+                        #num_variants => #enum_name::#variant_names(rand::Rng::gen(&mut rng)),
+                    )*
+                    _ => unreachable!()
+                }
+            }
+        };
+    }
+}
+
+fn derive_struct(item: DataStruct, name: Ident) -> proc_macro2::TokenStream {
+    if item.fields.iter().any(|f| f.ident.is_none()) {
+        unimplemented!("unit struct not implemented");
+    } else if item
+        .fields
+        .iter()
+        .any(|f| !matches!(f.vis, Visibility::Public(_)))
+    {
+        panic!("struct must have all public fields")
+    }
+
+    let fields_str = item
+        .fields
+        .iter()
+        .map(|field| field.ident.as_ref().map(|f| f.to_string()))
+        .collect::<Vec<_>>();
+
+    let macro_name = Ident::new(&format!("random_{}", name.to_string()), Span::call_site());
+    let mod_name =
+        Ident::new(&format!("mod_{}", name.to_string().to_lowercase()), Span::call_site());
+    quote::quote! {
+        pub use #mod_name::#macro_name;
+        #[macro_use]
+        mod #mod_name {
+            #[macro_export]
+            macro_rules! #macro_name {
+                ($(($field_name:ident, $value:expr)),*) => {
+                    {
+                        let mut rng = rand::thread_rng();
+                        let mut this: _ = rand::Rng::gen(&mut rng);
+                        let fields_to_get = std::collections::HashSet::from([#(#fields_str),*]);
+                        $(
+                            if fields_to_get.contains(std::stringify!($field_name)) {
+                                this.$field_name = $value;
+                            }
+                        )*
+                        this
+                    }
+                };
+            }
+            pub use #macro_name;
+        }
+    }
+}
```

Comment:
this is silly. just add a rand_default function that rand gens and then you can just do.

generating a marco for each of them is degen. just have it generate builder functions so that you can do a `rand_default().set_field1(v1).set_field_2(v2)`

Reply:
Is way more readable, and we aren't generating all of these macros that will just create a mess of the codebase